server/src/com/thoughtworks/go/config/GoConfigDao.java

```diff
@@ -221,13 +221,8 @@ public CruiseConfig update(CruiseConfig cruiseConfig) {
         };
     }
 
-    private UpdateConfigCommand environmentAdder(final BasicEnvironmentConfig environmentConfig) {
-        return new UpdateConfigCommand() {
-            public CruiseConfig update(CruiseConfig cruiseConfig) {
-                cruiseConfig.addEnvironment(environmentConfig);
-                return cruiseConfig;
-            }
-        };
+    private UpdateConfigCommand environmentAdder(final BasicEnvironmentConfig environmentConfig, Username user) {
```

Comment:
Please remove and inline these one line "factory" methods, we're trying to get away from these.